File: EcommerceBackend/EcommerceBackend.Domain/src/Entities/Product.cs
namespace EcommerceBackend.Domain.src.Entities
{
    public class Product : BaseEntity
    {
        public required string Name { get; set; }
        public required string Description { get; set; }
        public Decimal Price { get; set; }
        public Guid CategoryId { get; set; }
        public virtual Category Category { get; set; }
        public int Inventory { get; set; }
        public string ImageUrl { get; set; }
    }
}

File: EcommerceFrontend/frontend/src/type/Product.ts
import { Category } from "./Category"

export interface Product {
    id: string
    name: string 
    price: number
    description: string 
    category: Category
    inventory: number
    imageUrl: string
}

export interface ProductState {
    products: Product[]
    loading: boolean
    error: string | null
    singleProduct: Product | null
}

export interface CreateProduct {
    name: string
    description: string 
    price: number
    categoryId: string
    inventory: number
    imageUrl: string
}

export interface UpdateProduct {
    id: string
    name: string
    price: number
    description: string 
    categoryId: string
    inventory: number
    imageUrl: string
}

export interface ProductUpdate {
    id: string
    update: Partial<Omit<Product, "id">> & { id?: never }
}

export interface ProductProps {
    product: Product
}

export interface FileUploadResponse {
    originalname: string;
    filename: string;
    location: string;
}

export interface EditProductModalProps {
    open: boolean;
    product: UpdateProduct;
    onClose: () => void;
    onSubmit: (updatedProduct: UpdateProduct) => void;
}

export interface fetchSingleProductProps {
    loading: boolean
    error: string | null
    product: Product | null
}

export interface FavoriteButtonProps {
    favProduct: Product;
}